reframe/frontend/cli.py

```diff
@@ -537,12 +542,53 @@ def main():
     printer.debug(format_env(options.env_vars))
 
     # Setup the check loader
-    loader = RegressionCheckLoader(
-        load_path=site_config.get('general/0/check_search_path'),
-        recurse=site_config.get('general/0/check_search_recursive'),
-        ignore_conflicts=site_config.get('general/0/ignore_check_conflicts')
-    )
+    if options.retry_failed:
+        with open(options.retry_failed) as f:
+            try:
+                restart_report = json.load(f)
+            except json.JSONDecodeError as e:
+                raise ReframeFatalError(
+                    f"invalid runreport: '{restart_report}'"
+                ) from e
+
+        schema_filename = os.path.join(reframe.INSTALL_PREFIX, 'reframe',
+                                       'schemas', 'runreport.json')
+        with open(schema_filename) as f:
+            try:
+                schema = json.load(f)
+            except json.JSONDecodeError as e:
+                raise ReframeFatalError(
+                    f"invalid schema: '{schema_filename}'"
+                ) from e
 
+        try:
+            jsonschema.validate(restart_report, schema)
+        except jsonschema.ValidationError as e:
+            raise ValueError(f"could not validate restart runreport: "
+                             f"'{restart_report}'") from e
+
+        failed_checks = set()
+        failed_checks_prefixes = set()
+        # for run in restart_report['runs']:
+        for testcase in restart_report['runs'][-1]['testcases']:
+            if testcase['result'] == 'failure':
+                failed_checks.add(hash(testcase['name']) ^
+                                  hash(testcase['system']) ^
+                                  hash(testcase['environment']))
+                failed_checks_prefixes.add(testcase['prefix'])
+
+        loader = RegressionCheckLoader(
+            load_path=site_config.get('general/0/check_search_path'),  #failed_checks_prefixes,
```

Comment:
I started doing it like that, but later I found that if there's an incorrect test on that prefix, it is going to be loaded when reframe is discovering tests and it might crash. That happens for instance if trying to restart a test from `resources/checks_unlisted`.